utils/tests/test_fields.py

```diff
@@ -69,3 +73,324 @@ def test_generate_fake_data(self):
+        assert len([field for field in self.object_field.object_fields if field.name == "testSentence"]) == 1
+        assert unique_field in self.object_field.object_fields
+
+    def test_combine_fields(self):
+        """
+        Tests combining fields with this ObjectField.
+        Expected behavior:
+         - If the field is a subclass of BaseAPIField, but not another ObjectField, it should simply add the field to
+           this ObjectField's list of sub-fields.
+         - If the field is another ObjectField instance, it should add all of the sub-fields of the ObjectField to this
+           ObjectField's list of sub-fields.
+        """
+        self.object_field.object_fields = []
+
+        array_field = ArrayField(name="arrayField")
+        self.object_field.combine_fields(array_field)
+        assert array_field in self.object_field.object_fields
+
+        self.object_field.combine_fields(array_field, unique=True)
+        assert len([field for field in self.object_field.object_fields if field.name == "arrayField"]) == 1
+
+        test_object = ObjectField(name="testObject")
+        test_object_subfield = BaseAPIField(name="testSentence", data_type=DataType.SENTENCE)
+        test_object.add_field(test_object_subfield)
+
+        self.object_field.combine_fields(test_object)
+        assert test_object not in self.object_field.object_fields
+        assert all([test_field in self.object_field.object_fields for test_field in test_object.object_fields])
+
+        with pytest.raises(TypeError):
+            self.object_field.combine_fields("field string")
+
+    def test_get_field(self):
+        """ Tests retrieving a given field, by name, from this ObjectField's list of sub-fields. """
+        self.object_field.object_fields = []
+
+        base_field = BaseAPIField(name="baseField", data_type=DataType.INTEGER)
+        self.object_field.add_field(base_field)
+        returned_field = self.object_field.get_field("baseField")
+        assert returned_field == base_field
+
+    def test_update_required_fields(self):
+        """ Tests updating a list of fields to be 'required' """
+        self.object_field.object_fields = []
+
+        required_fields = ["baseField", "randomField", "phoneField", "spaceField"]
+
+        base_field = BaseAPIField(name="baseField", data_type=DataType.INTEGER)
+        random_field = ArrayField(name="randomField")
+        enum_field = EnumField(name="enumField")
+        space_field = ObjectField(name="spaceField")
+
+        self.object_field.add_fields([base_field, random_field, enum_field, space_field])
+
+        assert base_field.required is False
+        assert random_field.required is False
+        assert enum_field.required is False
+        assert space_field.required is False
+
+        self.object_field.update_required_fields(required_fields)
+
+        assert base_field.required is True
+        assert random_field.required is True
+        assert enum_field.required is False
```

Comment:
Why is this still false?

Reply:
Because its name wasn't in the `required_fields` list I passed in. Only these fields were going to be marked required: `["baseField", "randomField", "phoneField", "spaceField"]`. I set that at the top of this function, which I don't think is very readable. I'll change it to look like:

```python
        self.object_field.update_required_fields(["baseField", "randomField", "phoneField", "spaceField"])

        assert base_field.required is True
        assert random_field.required is True
        assert enum_field.required is False
        assert space_field.required is True
```

Hopefully that makes it more readable